### PR TITLE
feat: adds reference sketeton

### PR DESCRIPTION
**Pull request by killme2008**

### CODE REVIEW

**Comment by killme2008** [comment]
Still draft, don't merge it.

File: docs/reference/configuration.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+This page describes methods for configuring GreptimeDB server settings. Configuration can be set in TOML file.
+
+When a parameter is absent in the configuration file, then the default value is used.
+
+All sample configuration files are in project's [config](https://github.com/GreptimeTeam/greptimedb/tree/develop/config) folder.
+
+
+## Specify configuration file
+
+You can specify configuration file by command line arg `-c [file_path]`, for eaxmple:
```
Comment (apdong2022) [inline docs/reference/configuration.md:12]:
```suggestion
You can specify the configuration file by using the command line arg `-c [file_path]`, for example:
```

File: docs/reference/configuration.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
+# Configuration
+
+This page describes methods for configuring GreptimeDB server settings. Configuration can be set in TOML file.
+
+When a parameter is absent in the configuration file, then the default value is used.
+
+All sample configuration files are in project's [config](https://github.com/GreptimeTeam/greptimedb/tree/develop/config) folder.
```
Comment (apdong2022) [inline docs/reference/configuration.md:7]:
```suggestion
All sample configuration files are in the project's [config](https://github.com/GreptimeTeam/greptimedb/tree/develop/config) folder.
```

File: docs/reference/configuration.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+runtime_size = 2
+check_pwd = false
+```
+
+All these options are optional, the default values are list as above. If you want to disable some options such as disabling OpenTSDB protocol support, you can just remove the `prometheus_options` or set it's `enable` value to be `false`.
+
+
+## Standalone
+
+When you start the GreptimeDB in standalone mode, you can configure it as below:
```
Comment (apdong2022) [inline docs/reference/configuration.md:53]:
```suggestion
When you use GreptimeDB in the standalone mode, you can configure it as below:
```

**Comment by apdong2022** [comment]
53: can we use initiate or use for "start"? start Greptime DB sounds a bit vague

**Comment by killme2008** [comment]
@apdong2022  Thanks for your review, can you review it again, I add more descriptions about the configuration options.

I will resolve all these conversations after your review. Thank you.

File: docs/reference/configuration.md [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+|                    | enable       | Boolean | Whether to enable OpenTSDB protocol in HTTP API, true by default                |
+|                    | addr         | String  | OpenTSDB telnet API server address, "0.0.0.0:4242" by default                   |
+|                    | runtime_size | Integer | The number of server worker threads, 2 by default                               |
+| prometheus_options |              |         | Prometheus protocol options                                                     |
+|                    | enable       | Boolean | Whether to enable Prometheus remote write and read in HTTP API, true by default |
+| postgres_options   |              |         | PostgresSQL server options                                                      |
+|                    | addr         | String  | Server address, '0.0.0.0:4003' by default                                       |
+|                    | runtime_size | Integer | The number of server worker threads, 2 by default                               |
+|                    | check_pwd    | boolean | Wheter to check password, it's not supported right now, always false.           |
```
Comment (shuiyisong) [inline docs/reference/configuration.md:72]: typo. Wheter should be whether

Comment (killme2008) [reply docs/reference/configuration.md:72]: Good catch!

**Comment by killme2008** [comment]
I think we can merge this PR first, I am still working on other docs. @apdong2022 @MichaelScofield

**Review by apdong2022 — APPROVED** [review]

**Review by evenyag — APPROVED** [review]
LGTM

File: docs/reference/configuration.md [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+server_addr = '127.0.0.1:3002'
+store_addr = '127.0.0.1:2379'
+datanode_lease_secs = 30
+```
+
+| Key                 | Type    | Description                                                                                                             |   |
+|---------------------|---------|-------------------------------------------------------------------------------------------------------------------------|---|
+| bind_addr           | String  | The bind address of metasrv, '127.0.0.1:3002' by default.                                                                 |   |
+| server_addr         | String  | The communication server address for frontend and datanode to connect metasrv,  '127.0.0.1:3002' by default for localhost |   |
+| store_addr          | String  | Etcd server address, '127.0.0.1:2379' by default                                                                        |   |
```
Comment (waynexia) [inline docs/reference/configuration.md:209]:
```suggestion
| store_addr          | String  | etcd server address, '127.0.0.1:2379' by default                                                                        |   |
```

File: docs/reference/configuration.md [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+
+```toml
+bind_addr = '127.0.0.1:3002'
+server_addr = '127.0.0.1:3002'
+store_addr = '127.0.0.1:2379'
+datanode_lease_secs = 30
+```
+
+| Key                 | Type    | Description                                                                                                             |   |
+|---------------------|---------|-------------------------------------------------------------------------------------------------------------------------|---|
```
Comment (waynexia) [inline docs/reference/configuration.md:206]:
```suggestion
|---------------------|---------|-------------------------------------------------------------------------------------------------------------------------|
```

Looks like there is an empty column in the end. Other rows should be change correspondingly.

**Pull request merged**